### ISSUE TITLE
Allow custom backup file suffixes instead of having to end with .zip.aes.

### ISSUE THREAD

**Issue by Andiedie** [duplicati/duplicati#5601]
Now, when I open the zip and aes encryption, the file name suffix must be xxx.zip.aes

I hope to customize the file name suffix, so that regardless of the options enabled, the suffix is always xxx.zip

---

The reason I want to do this is that I hope to back up data to a remote WebDAV, but this service provider has some restrictions and does not allow uploading files with unknown suffixes, including .aes.

**Comment by kenkendk** [comment]
This is not a simple thing because Duplicati uses the suffix to determine file types. Without a reliable suffix, we would need to change the logic to use content-sniffing where we peek at a few bytes to guess what file format it is.

**Comment by Andiedie** [comment]
I have a suggested implementation (but since I don't have a deep understanding of the code, it may be very naive; please advise).
- For a backup configuration, if no specific suffix is configured, then the current logic remains unchanged;
- if a specific suffix is configured, then when writing to the target storage, this suffix should be added (instead of replacing to reduce the difficulty and errors of implementation);
- when reading for recovery from the target storage, this suffix should be removed.

This seems to be a simple implementation and will not cause destructive changes.

**Comment by kenkendk** [comment]
Yes, that could be done transparently. It would be a general "rename" layer for the names. I think it could be implemented as a backend that wraps another backend and translates the names.

The remote names needs to be translated in the methods: `put`, `get`, `list` and `delete`.  The rest can just pass-through to the underlying backend.
